packages/@dcl/ecs/src/engine/index.ts

```diff
@@ -202,23 +202,31 @@ function preEngine(): PreEngine {
 
   const Transform = components.Transform({ defineComponentFromSchema })
 
-  function* getTreeEntityArray(firstEntity: Entity, proccesedEntities: Entity[]): Generator<Entity> {
+  function getComponentEntityTree<T>(
+    entity: Entity,
+    component: ComponentDefinition<T & { parent?: Entity }>
+  ): Generator<Entity> {
+    const entities = new Map(getEntitiesWith(component))
+    return genEntityTree(entity, entities)
+  }
+
+  function* genEntityTree<T>(entity: Entity, entities: Map<Entity, T & { parent?: Entity }>): Generator<Entity> {
```

Comment:
This genEntityTree does not ensure pre-order in-order or post-order

Reply:
nono, it just returns all the entities involved in the tree, in any order. We could improve it to return different traversing orders, but I think for now it's enough, since we are only looking for a list of entities to apply some logic to.